File: app/models/question.rb
# == Schema Information
#
# Table name: questions
#
#  id               :bigint(8)        not null, primary key
#  description      :text             not null
#  feedback         :text
#  points_available :integer          default(1), not null
#  created_at       :datetime         not null
#  updated_at       :datetime         not null
#  test_id          :integer          not null
#

class Question < ApplicationRecord
  belongs_to :test
  has_many :answers, dependent: :destroy
  has_many :referee_answers, dependent: :destroy
end

File: dotnet/ManagementHub.Models/Question.cs
using System;
using System.Collections.Generic;

namespace ManagementHub.Models
{
    public partial class Question
    {
        public Question()
        {
            Answers = new HashSet<Answer>();
            RefereeAnswers = new HashSet<RefereeAnswer>();
        }

        public long Id { get; set; }
        public long TestId { get; set; }
        public string Description { get; set; } = null!;
        public int PointsAvailable { get; set; }
        public string? Feedback { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        public virtual Test Test { get; set; } = null!;
        public virtual ICollection<Answer> Answers { get; set; }
        public virtual ICollection<RefereeAnswer> RefereeAnswers { get; set; }
    }
}
